### ISSUE TITLE
Open links without focusing firefox

### ISSUE THREAD

**Issue by noflag** [lwindolf/liferea#106]
It gets annoying when there are a lot of links I'm clicking in an article and the firefox window keeps popping up.

**Comment by lwindolf** [comment]
The external browser focussing properties are influenced both by the window manager settings and the browser itself. A launching application cannot really control it. The way Liferea launches browsers already tries to pass those parameters that cause a background launch.

Experience shows that launching in tabs more often stays in background than launching new windows. You might want to check into this.

**Comment by asl97** [comment]
i know this issue is old but i thought of another solution,  just mark liferea's window with `always on top`

**Comment by lwindolf** [comment]
@asl97 It would solve the bug and at the same time cause uproar with a lot of users suddenly having an unwanted behaviour.

It doesn't matter though as the bug is fixed now.

**Comment by asl97** [comment]
@lwindolf i mean it as an advice for @noflag.

**Comment by noflag** [comment]
thanks, can't believe i didn't think of this

**Comment by erness** [comment]
set browser.tabs.loadDivertedInBackground to true in about:config

**Comment by lwindolf** [comment]
@antistress I understand you issue, but I do not plan to change the implementation. Nowadays Liferea uses the only safe way to open links the GTK method `gtk_show_uri_on_window()` that works for non-sandboxed and sandboxed apps.

As it is a GTK method the application has no control over the semantics. So the behaviour cannot be changed AFAIK.

For use cases where this does not match Liferea has the custom browser command preference where you can launch Firefox in a way that you prefer.

So to summarize: the introduction of the sandboxing (Snap and Flatpak) further limited what app authors can do for you in 2025 :-(

**Comment by antistress** [comment]
Ok, thanks !

**Comment by antistress** [comment]
> I've noticed that Thunderbird does open link in the background of my Firefox whereas Liferea doesn't. > 
> My setup: Debian GNOME with Wayland Flatpak Firefox 134.0.1 Flatpak Thunderbird 128.6.0esr Flatpak Liferea 1.14.6

Ok, here is an interesting update
I had to switch back **from** flatpack versions of firefox and thunderbird **to** deb packages
(I now run thunderbird ESR debian deb package 128.11.0esr and Firefox release mozilla deb package 139.0.4 [1])

I've noticed that Thunderbird does not open link in the background of my Firefox anymore !

For what it's worth.

Thanks !

{1] https://support.mozilla.org/fr/kb/installer-firefox-linux#w_installation-par-paquet-deb-pour-les-distributions-basees-sur-debian-recommande

**Comment by lwindolf** [comment]
@antistress Thanks for the this info!
